cpp/code/CMakeLists.txt

```
@@ -42,9 +48,14 @@ function(RECIPE TARGET)
             common.cc
             main.cc
     )
+    if(${ARROW_VERSION} VERSION_GREATER "9.0.0")
```

```suggestion
    if(TARGET Arrow::arrow_shared)
```

In general, feature based check is better than version check.